File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/common/SourceBasedActorState.java
package org.pyjjs.scheduler.core.api.impl.actors.common;

import akka.actor.ActorContext;

public class SourceBasedActorState<T> extends ActorState {

    private T source;

    public SourceBasedActorState(ActorContext actorContext) {
        super(actorContext);
    }

    public T getSource() {
        return source;
    }

    public void setSource(T source) {
        this.source = source;
    }

    @Override
    protected ActorState copySelf() {
        return this;
    }
}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/common/SourceBasedActorState.kt
package org.pyjjs.scheduler.core.api.impl.actors.common

import akka.actor.ActorContext
import org.pyjjs.scheduler.core.model.IdentifiableObject

open class SourceBasedActorState<T: IdentifiableObject>(actorContext: ActorContext) : ActorState(actorContext) {
    lateinit var source: T
    override fun copySelf() = this
}
